### PR TITLE
Cafeteria Scheduler

### PR DESCRIPTION
**Pull request by Tcintra**
Implement a rudimentary cafeteria scheduler that, given lunch and dinner timeranges, a meal timerange, and a list of students, returns a schedule object with students grouped into meal blocks based upon their preferences.

### CODE REVIEW

**Comment by christinachance** [comment]
Add new lines at end of files

File: project/src/main/java/com/google/sps/classes/CafeteriaScheduler.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        unhappyDinnerStudents.add(student);
+      }
+    }
+
+    while (!unhappyDinnerStudents.isEmpty()) {
+      for (Block block : dinnerBlocks) {
+        while (block.getCapacity() < maxCapacity && !unhappyDinnerStudents.isEmpty()) {
+          Student student = unhappyDinnerStudents.get(0);
+          block.addStudent(student);
+          unhappyDinnerStudents.remove(student);
```
Comment (christinachance) [inline project/src/main/java/com/google/sps/classes/CafeteriaScheduler.java]: This seem repetitive and very similar to the scheduling for lunch time. Is there a way that you can make a helper function that you can feed in the time of the day for?

File: project/src/main/java/com/google/sps/classes/Block.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import com.google.sps.classes.TimeRange;
+
+public final class Block {
+  
+  private List<Student> students;
+  private final TimeRange time;
+  private final int duration;
+
+  public Block(int start, int duration) {
+
```
Comment (rivera-fernando) [inline project/src/main/java/com/google/sps/classes/Block.java]: You have additional whitespace that I don't think you really need? Just makes your file a lot longer than necessary.

File: project/src/main/java/com/google/sps/classes/CafeteriaScheduler.java (hunk 1 above)
Comment (Tcintra) [reply project/src/main/java/com/google/sps/classes/CafeteriaScheduler.java]: Yeah, I was planning on doing that today! Thanks for the input

File: project/src/main/java/com/google/sps/classes/Block.java (hunk 2 above)
Comment (Tcintra) [reply project/src/main/java/com/google/sps/classes/Block.java]: True I'm gonna trim it, not sure why the formatter did that.

**Review by christinachance — APPROVED** [review]

**Review by rivera-fernando — APPROVED** [review]

**Review by serzhk — APPROVED** [review]

**Pull request merged**